### PR TITLE
optee-utee: support setting capacity of shared buffer of LoadablePlugin

### PR DESCRIPTION
**Pull request by ivila**
1) introduce a Command struct for flexible use
2) allow developers settings capacity of shared buffer when invoking
   command

### CODE REVIEW

**Comment by ivila** [comment]
Found some error during CI, I am trying to reproduce on my machine (I think in some platform the c_char was treated as i8 instead of u8)

**Comment by ivila** [comment]
Pipeline fixed: https://github.com/ivila/incubator-teaclave-trustzone-sdk/actions/runs/12194856005/job/34019402495

**Comment by ivila** [comment]
This PR relates to #153

File: optee-utee/src/extension.rs [hunk 1]
```diff
@@ -26,32 +26,304 @@ pub struct LoadablePlugin {
...
+    buffer: Vec<u8>,
+}
+
 impl LoadablePlugin {
     pub fn new(uuid: &Uuid) -> Self {
         Self { uuid: uuid.to_owned() }
     }
-    pub fn invoke(&mut self, command_id: u32, subcommand_id: u32, data: &[u8]) -> Result<Vec<u8>> {
-        let raw_uuid: Uuid = self.uuid;
+    pub fn invoke(&self, command_id: u32, subcommand_id: u32, data: &[u8]) -> Result<Vec<u8>> {
```
Comment (DemesneGH) [inline optee-utee/src/extension.rs:40]: Can we have some comments on each functions to provide guidance for developers, such as the suitable scenario for `invoke()` and `invoke_with_capacity()`? The comments can also be rendered into our API documentation. 
FYI One example for the comment format: https://github.com/apache/incubator-teaclave-trustzone-sdk/blob/20e698863932d395b48002e5fb126491e0caef0d/optee-utee/src/object.rs#L83-L94

File: optee-utee/src/extension.rs [hunk 2]
```diff
@@ -26,32 +26,304 @@ pub struct LoadablePlugin {
...
-       
+    }
+}
+
+impl<'a> LoadablePluginCommand<'a> {
+    fn new_with_capacity(plugin: &'a LoadablePlugin, cmd_id: u32, capacity: usize) -> Self {
+        Self {
+            plugin,
+            cmd_id,
+            sub_cmd_id: 0,
```
Comment (DemesneGH) [inline optee-utee/src/extension.rs:111]: `new_with_capacity()` set a default sub_cmd_id then we set that manually:
```
        self.new_command_with_capacity(command_id, capacity)
            .sub_cmd(subcommand_id)
```
How about setting the sub_cmd when new()?
```
new_with_capacity(plugin: &'a LoadablePlugin, cmd_id: u32, sub_cmd_id: u32, capacity: usize)
```

File: optee-utee/src/extension.rs [hunk 3]
```diff
@@ -26,32 +26,304 @@ pub struct LoadablePlugin {
...
             )
         } {
             raw::TEE_SUCCESS => {
-                assert!(outlen <= (data.len()));
-                let mut outbuf = vec![0; outlen];
-                outbuf.copy_from_slice(&data[..outlen]);
-                
-                Ok(outbuf)
-            },
+                assert!(outlen <= buffer.len());
```
Comment (DemesneGH) [inline optee-utee/src/extension.rs:97]: This is a legacy issue that we use `assert!` which may cause panic. Please help to correct this to raise an error for developer, thanks!

**Comment by ivila** [comment]
Copy, I will update it later(before next tuesday)

**Review by DemesneGH — APPROVED** [review]

**Comment by DemesneGH** [comment]
`Reviewed-by: Yuan Zhuang <yuanz@apache.org>`

**Comment by ivila** [comment]
> `Reviewed-by: Yuan Zhuang <yuanz@apache.org>`

I already add it in my first commit😈

**Review by mssun — APPROVED** [review]
LGTM, thanks!

**Review by AI-Memory — APPROVED** [review]

**Comment by DemesneGH** [comment]
@b49020 What do you think?

**Comment by ivila** [comment]
> `Acked-by: Sumit Garg <sumit.garg@linaro.org>`

@b49020 done, thanks.

**Pull request merged**